sycl/test-e2e/SYCLBIN/simple_kernel_aot.cpp

```diff
@@ -0,0 +1,88 @@
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS-NEXT:   Metadata:
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS:      Number of IR Modules: 0
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS-NEXT: Number of Native Device Code Images: 1
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS-NEXT: Native device code image 0:
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS-NEXT:   Metadata:
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS-NEXT:     SYCLBIN/native device code image metadata:
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS-NEXT:        arch:{{.*}}
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS-NEXT:        target:{{.*}}spir64_gen-unknown
+// CHECK-INPUT-ALL-GPU-DEVICE-ARCHS-NEXT:   Raw native device code image bytes: <Binary blob of {{.*}} bytes>
+
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH:      Version: {{[1-9]+}}
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT: Global metadata:
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT:   SYCLBIN/global metadata:
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT:     state: 1
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT: Number of Abstract Modules: 1
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT: Abstract Module 0:
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT:   Metadata:
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH:      Number of IR Modules: 0
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT: Number of Native Device Code Images: 1
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT: Native device code image 0:
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT:   Metadata:
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT:     SYCLBIN/native device code image metadata:
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT:        arch:{{.*}}
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT:        target:{{.*}}spir64_gen-unknown
+// CHECK-OBJECT-BMG-G21-GPU-DEVICE-ARCH-NEXT:   Raw native device code image bytes: <Binary blob of {{.*}} bytes>
+
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH:      Version: {{[1-9]+}}
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT: Global metadata:
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT:   SYCLBIN/global metadata:
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT:     state: 2
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT: Number of Abstract Modules: 1
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT: Abstract Module 0:
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT:   Metadata:
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH:      Number of IR Modules: 0
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT: Number of Native Device Code Images: 1
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT: Native device code image 0:
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT:   Metadata:
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT:     SYCLBIN/native device code image metadata:
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT:        arch:{{.*}}
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT:        target:{{.*}}spir64_gen-unknown
+// CHECK-EXECUTABLE-BMG-G21-GPU-DEVICE-ARCH-NEXT:   Raw native device code image bytes: <Binary blob of {{.*}} bytes>
```

Comment:
I think it would be fine to have this in sycl/test instead.

Reply:
The only reason I see not putting it to sycl/test folder is that the test has a dependency on NEO - especially on `ocloc` binary, and sycl/test tests should be device-agnostic (no backend runtimes should be present in the environment)

Reply:
> > Let's try, should work. Still investigating though why it doesn't work in CI - as it works locally. The binary now in the install folder, but still it didn't fix the issue, probably need to add to some other place in CMake files.
> 
> Maybe we need to do something like in https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/lit.cfg.py#L627?

Thanks! It works now :)